packages/runner/src/Runner.ts

```diff
@@ -1,3 +1,17 @@
+export type RunnerListenerCallback<ARG extends any[]> = (...args: ARG) => unknown;
+
+export type ValidRunnerItem<T extends string, ARG extends unknown[] = any[]> =
+    { [K in T]: RunnerListenerCallback<ARG> | unknown };
+
+export type AnyRunnerItem = Record<string, unknown>;
+
+export type EmptyRunnerItem = Record<string, never>;
+
+export type RunnerItem<T, ARG extends unknown[] = any[]> =
+    T extends string ?
+        ValidRunnerItem<T, ARG> & AnyRunnerItem | EmptyRunnerItem :
+        unknown;
```

Comment:
Small suggestion to prefix all the types with "RunnerItem". Make them alphabetically sort together with auto-complete and in docs.

Also, is it necessary to export the Empty and Any types? Why not just inline those?

```suggestion
export type RunnerItemValid<T extends string, ARG extends unknown[] = any[]> =
    { [K in T]: RunnerListenerCallback<ARG> | unknown };

export type RunnerItemAny = Record<string, unknown>;

export type RunnerItemEmpty = Record<string, never>;

export type RunnerItem<T, ARG extends unknown[] = any[]> =
    T extends string ?
        RunnerItemValid<T, ARG> & RunnerItemAny | RunnerItemEmpty :
        unknown;
```

Reply:
Good point! Fixed here: https://github.com/pixijs/pixijs/pull/9948/commits/cc4a6687435d848b83d3ea2e4facdf6978ccc381
I've removed exports of RunnerItemValid, RunnerItemAny, RunnerItemEmpty and left only a generic RunnerItem.

Usage:

```typescript
const runner = new Runner('update');

let item: RunnerItem<'update'>;

item = {
    update: () => console.log('updated')
};
runner.add(item);

// error: 'update' key is not present
item = {
    destroy: () => console.log('destroyed')
};
```